build.py

```diff
@@ -43,27 +39,20 @@ def find_mounted_directories(build_dir: Path) -> List[Path]:
 
 
 def cleanup(build_dir):
-    dirs = find_mounted_directories(build_dir)
-    for d in dirs:
-        subprocess.run(
-            ["umount", d],
-        )
+    print("Syncing disks")
+    os.sync()
+    print("Unmounting")
+    subprocess.run(
+        ["umount", "-r", build_dir],
```

Comment:
```suggestion
        ["umount", "--recursive", build_dir],
```
From the umount man page:
```
       -R, --recursive
              Recursively  unmount each specified directory.  Recursion for each directory will stop if any unmount
              operation in the chain fails for any reason.  The relationship between mountpoints is  determined  by
              /proc/self/mountinfo  entries.   The filesystem must be specified by mountpoint path; a recursive un‐
              mount by device name (or UUID) is unsupported.

       -r, --read-only
              When an unmount fails, try to remount the filesystem read-only.
```

Reply:
This would explain some things